doc/rst/language/evolution.rst

```diff
@@ -14,6 +14,40 @@ The purpose of this flag is to identify portions of a program that use a
 language or library feature has recently changed meaning or which is
 expected to change meaning in the future.
 
+version 1.32, September 2023
+----------------------------
+
+Version 1.32 deprecates the ``c_string`` type in user interfaces. Please
+replace occurrences of ``c_string`` with ``c_ptrConst(c_char)``. Note that you
+need to ``use`` or ``import`` the ``CTypes`` module.
+
+Here are some cases where directly replacing ``c_string`` with
+``c_ptrConst(c_char)`` may not work and what to do instead:
+
+==================================  ============================================
+if your code is...                  update it to...
+==================================  ============================================
+casting ``c_string`` to ``string``  use a ``string.create*Buffer()`` method
+casting ``c_string`` to ``bytes``   use a ``bytes.create*Buffer()`` method
+casting ``c_string`` to other type  create a string and cast it to other type
+casting ``string`` to ``c_string``  replace cast with ``.c_str()``
+casting ``bytes`` to ``c_string``   replace cast with ``.c_str()``
+casting other type to ``c_string``  create a string and call ``.c_str()`` on it
+using ``param c_string``            use ``param string``
```

Comment:
"use ``param string`` and call ``.c_str()`` as needed"?

Reply:
Thinking about it some more, I'm not sure it would help to call `.c_str()` because the result would not be a `param`.
I don't expect much, if any, use of `param c_string` in the wild, so this is just to make it very clear that `string` would be the appropriate type to use for a `param` replacement. I'm not sure what even would be the use case for `param c_string` in a Chapel program, and if someone was using it this should be the final push to restructure their code.

Reply:
Okay that makes sense. Thanks for clarifying!